src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/EncryptedReference.cs

```diff
@@ -1,33 +1,34 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Diagnostics.CodeAnalysis;
 using System.Xml;
 
 namespace System.Security.Cryptography.Xml
 {
     public abstract class EncryptedReference
     {
-        private string _uri;
-        private string _referenceType;
-        private TransformChain _transformChain;
-        internal XmlElement _cachedXml;
+        private string? _uri;
+        private string? _referenceType;
+        private TransformChain? _transformChain;
+        internal XmlElement? _cachedXml;
 
         protected EncryptedReference() : this(string.Empty, new TransformChain())
         {
         }
 
-        protected EncryptedReference(string uri) : this(uri, new TransformChain())
+        protected EncryptedReference(string? uri) : this(uri, new TransformChain())
         {
         }
 
-        protected EncryptedReference(string uri, TransformChain transformChain)
+        protected EncryptedReference(string? uri, TransformChain transformChain)
         {
             TransformChain = transformChain;
             Uri = uri;
             _cachedXml = null;
         }
 
-        public string Uri
+        public string? Uri
```

Comment:
Since the setter here throws for null, this property should be declared as

```suggestion
        [DisallowNull]
        public string? Uri
```

There are probably others in the same boat.

Reply:
Done. But I get `error : CannotRemoveAttribute : Attribute 'System.Diagnostics.CodeAnalysis.DisallowNullAttribute' exists on parameter 'value' on member 'System.Security.Cryptography.Xml.EncryptedReference.Uri.set(System.String)' in the implementation but not the reference.`